### PR TITLE
Discount api id from subscription

### PR DESCRIPTION
**Pull request by rupertbates**
## What does this change?
Previously the discount-api expected to be told which discount to apply to the given subscription. This PR changes that so that it works out the appropriate discount to apply based on the billing period of the sub.

### CODE REVIEW

File: handlers/discount-api/src/productToDiscountMapping.ts [hunk 1]
```diff
@@ -1,16 +1,47 @@
...
 export const ProductToDiscountMapping: {
-	[K in Stage]: Array<{
-		discountProductRatePlanId: string;
-		name: string;
-		effectiveStartDate: string;
-		effectiveEndDate: string;
-		eligibleProductRatePlanIds: string[];
-	}>;
+	[K in Stage]: {
+		Month: {
```
Comment (andrewHEguardian) [inline handlers/discount-api/src/productToDiscountMapping.ts:27]: What about `Quarter`, which I think digisub can be?

File: handlers/discount-api/src/endpoints/previewDiscountEndpoint.ts [hunk 2]
```diff
@@ -21,13 +29,41 @@ export const previewDiscountEndpoint = async (
...
 	body: string | null,
 ) => {
+	const zuoraClient = await ZuoraClient.create(stage);
+	const catalog = await getZuoraCatalog(stage);
+	const eligibilityChecker = new EligibilityChecker(catalog);
+
 	const applyDiscountBody = applyDiscountSchema.parse(
 		JSON.parse(checkDefined(body, 'No body was provided')),
 	);
-	const preview = await PreviewDiscount.create(stage);
```
Comment (andrewHEguardian) [inline handlers/discount-api/src/endpoints/previewDiscountEndpoint.ts:27]: just curious why you got rid of the `PreviewDiscount` class?

File: handlers/discount-api/src/productToDiscountMapping.ts (hunk 1 above)
Comment (rupertbates) [reply handlers/discount-api/src/productToDiscountMapping.ts:27]: Oh yes, good spot 👍

File: handlers/discount-api/src/productToDiscountMapping.ts [hunk 3]
```diff
@@ -70,6 +76,16 @@ export const ProductToDiscountMapping: {
 				'2c92a0fb4edd70c8014edeaa4e8521fe',
 			],
 		},
+		Quarter: {
+			discountProductRatePlanId: '8a128adf8b64bcfd018b6b6fdc7674f5',
+			name: 'Cancellation Save Discount - 25% off for 12 months',
```
Comment (andrewHEguardian) [inline handlers/discount-api/src/productToDiscountMapping.ts:81]:
```suggestion
			name: 'Cancellation Save Discount - 25% off for 3 months',
```

Comment (rupertbates) [reply handlers/discount-api/src/productToDiscountMapping.ts]: Oops, got the discounts the wrong way round!

**Review by andrewHEguardian — APPROVED** [review]
Looks good to me 👍

**Pull request merged**